### PR TITLE
Update the validation string for date questions

### LINKED ISSUE #3078: Date validation error messages suggests using an incorrect date format
**Describe the bug**
After entering an invalid date and clicking next, the error message that pops up suggests using "YYYY/MM/DD", but the input field expects mm/dd/yyyy.

A few nuances to consider when solving this bug:
* I believe the placeholder text in `<input type="date">` is dependent on the language the user sets their browser to. (Based on this [stackoverflow](https://stackoverflow.com/questions/61186908/how-to-change-input-type-date-calender-language-localization) and linked [docs](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Input#localization)). So this is not always going to be mm/dd/yyyy as shown in my screenshot.
* We can set the `lang` attribute for the input tag to force a certain language (e.g. the one selected by the user within CiviForm).
  
My proposed solution (open to alternatives):
* Set `lang="en"` to ensure the format displayed to the user is always mm/dd/yyyy.
* Update all [message translations](https://github.com/civiform/civiform/blob/c337065f5e9cb0945f81e3b5ca9c952c05e05aca/server/conf/messages#L258) to use "MM/DD/YYYY".

I feel like this is the easiest solution to ensure the error message matches the input field format. 

Alternatively, we could try to fully localize by setting `lang` to the CiviForm selected language, but this introduces room for bugs where the translation we have may not match the input date format. Also, I think applicants are used to and expect the mm/dd/yyyy format, since they are US based.

**To Reproduce**

1. Answer a date question with invalid date 05/05/999999
2. Click "Save and Next"
3. Look at the error text!

**Screenshots**
Screen Shot 2022-08-05 at 2 04 29 PM

### PR DESCRIPTION
**Pull request by greg-hyde**
### Description

Update the validation string for date questions to not reference a specific date format

## Release notes

The title of the pull request will be used as the default release notes description. If more detail is needed to communicate to partners the scope of the PR's changes, use this release notes section.

#### General

Read the full guidelines for PRs [here](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#creating-a-pull-request)

- [x] Added the correct label: < feature | enhancement | bug | dependencies | infrastructure | ignore-for-release | database >
- [x] Assigned to a specific person, `civiform/developers`, or a [more specific round-robin list](https://github.com/civiform/civiform/wiki/Technical-contribution-guide#adding-reviewers).
- [x] Performed manual testing (Chrome and Firefox if it includes front-end changes)

#### User visible changes

- [x] Followed steps to [internationalize new strings](https://github.com/civiform/civiform/wiki/Internationalization-%28i18n%29#internationalization-for-application-strings)
- [x] Tested on mobile view. See [mobile device mode](https://developer.chrome.com/docs/devtools/device-mode/)
- [x] Manually tested at 200% size
- [x] Manually evaluated tab order

### Issue(s) this completes

Fixes #3078

### CODE REVIEW

**Comment by greg-hyde** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #6460** <a href="https://app.graphite.dev/github/pr/civiform/civiform/6460?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/civiform/civiform/6460?utm_source=stack-comment).

File: server/app/services/MessageKey.java [hunk 1]
```diff
@@ -130,12 +130,11 @@ public enum MessageKey {
   ERROR_NOT_FOUND_DESCRIPTION("error.notFoundDescription"),
   ERROR_NOT_FOUND_DESCRIPTION_LINK("error.notFoundDescriptionLink"),
   DATE_VALIDATION_INVALID_DATE_FORMAT("validation.invalidDateFormat"),
-  DATE_VALIDATION_MISFORMATTED("validation.dateMisformatted"),
   DAY_LABEL("label.day"),
+  DIALOG_DELETE_CONFIRMATION("dialog.deleteConfirmation"),
   DOB_EXAMPLE("label.dobExample"),
   DOB_ERROR_LABEL("label.errorDOB"),
   DOB_LABEL("label.dob"),
-  DIALOG_DELETE_CONFIRMATION("dialog.deleteConfirmation"),
```
Comment (rockycodes) [inline server/app/services/MessageKey.java:138]: question (blocking): did you mean to delete this as well?

File: server/conf/i18n/messages.am [hunk 2]
```diff
@@ -369,7 +369,6 @@ validation.phoneMustBeLocalToCountry=ያቀረቡት ስልክ የአገሩ አ
 # DATE QUESTION - text shown when answering a question where a user must select date. #
 #----------------------------------------------------------------------------------------------------------#
 
-validation.dateMisformatted=እባክዎ በዓዓዓዓ/ወወ/ቀቀ ቅርጸት ቀን ያስገቡ።
```
Comment (rockycodes) [inline server/conf/i18n/messages.am:372]: question (blocking): do you need to update the translated files or should that happen automatically via transifex? I know additions are automatic, but i'm not sure what happens with deletions.

**Review by rockycodes — CHANGES_REQUESTED** [review]
a couple of questions

File: server/app/services/MessageKey.java (hunk 1 above)
Comment (greg-hyde) [reply server/app/services/MessageKey.java:138]: this one just got moved above in the interest of keeping this alphabetized

File: server/conf/i18n/messages.am (hunk 2 above)
Comment (greg-hyde) [reply server/conf/i18n/messages.am:372]: I believe that they have to be deleted manually (see https://github.com/civiform/civiform/pull/7059 for another example)

File: server/app/services/MessageKey.java (hunk 1 above)
Comment (rockycodes) [reply server/app/services/MessageKey.java:138]: Ohh I see that now.

**Review by rockycodes — APPROVED** [review]

**Pull request merged**